crates/dojo-world/src/manifest_test.rs

```diff
@@ -209,39 +209,131 @@ fn parse_deployed_contracts_events_with_upgrade() {
         EmittedEvent {
             data: vec![felt!("0x66"), felt!("0x123")],
             keys: vec![],
-            block_number: 2,
+            block_number: Some(2),
             block_hash: Default::default(),
             from_address: Default::default(),
             transaction_hash: Default::default(),
         },
         EmittedEvent {
             data: vec![felt!("0x69"), felt!("0x123")],
             keys: vec![],
-            block_number: 9,
+            block_number: Some(9),
             block_hash: Default::default(),
             from_address: Default::default(),
             transaction_hash: Default::default(),
         },
         EmittedEvent {
             data: vec![felt!("0x77"), felt!("0x123")],
             keys: vec![],
-            block_number: 5,
+            block_number: Some(5),
             block_hash: Default::default(),
             from_address: Default::default(),
             transaction_hash: Default::default(),
         },
         EmittedEvent {
             data: vec![felt!("0x88"), felt!("0x789")],
             keys: vec![],
+            block_number: Some(5),
+            block_hash: Default::default(),
+            from_address: Default::default(),
+            transaction_hash: Default::default(),
+        },
+    ];
+
+    let actual_contracts = parse_contracts_events(deployed_events, upgrade_events);
+    similar_asserts::assert_eq!(actual_contracts, expected_contracts);
+}
+
+#[test]
+fn events_without_block_number_arent_parsed() {
```

Comment:
added a new test to ensure that events without block number arent parsed when fetching remote world states